crates/driver/src/domain/competition/solution/mod.rs

```diff
@@ -119,6 +120,64 @@ impl Solution {
         &self.score
     }
 
+    /// Converts the solution into an onchain settleable form and semantic, as
+    /// expected by the settlement contract.
+    ///
+    /// Skips JIT orders.
+    pub fn scorable_trades(&self) -> Result<Vec<settled::Trade>, SolutionError> {
+        let mut trades = Vec::with_capacity(self.trades.len());
+        for trade in self.user_trades() {
+            // Solver generated fulfillment does not include the fee in the executed amount
+            // for sell orders.
+            let executed = match trade.order().side {
+                order::Side::Sell => (trade.executed().0 + trade.fee().0).into(),
+                order::Side::Buy => trade.executed(),
+            };
+            let uniform_prices = settled::CustomClearingPrices {
```

Comment:
Note, that for uniform clearing prices you already introduced a type here:

https://github.com/cowprotocol/services/blob/e29c422b8bb02c8af0fef81e8554cbb554052c12/crates/driver/src/domain/competition/solution/trade.rs#L236-L242

Reply:
good catch, reused.